contracts/ERC20SimpleSwap.sol

```diff
@@ -38,6 +39,8 @@ contract SimpleSwap {
     uint canBeDecreasedAt; /* point in time after which harddeposit can be decreased*/
   }
 
+  /* The token against which this chequebook writes cheques */
+  ERC20 public token;
```

Comment:
I wonder if it makes sense to include the token address into the cheque. (Although as long as there is no factory on another chain on the same address with a different token that should not be an issue)

Reply:
We talked about it in person and decided that it should be solved by deploying the factory by CREATE2.